### PR TITLE
Implemented the fix for Calendar2 weekdays style for some Chinese languages

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] Component `README.md`
+ [ ] Tests

If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)

### CODE REVIEW

File: backpack-common/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt [hunk 1]
```diff
@@ -152,3 +152,13 @@ enum class CellStatusStyle {
      */
     Label,
 }
+
+private fun findBestWeekdayStyleForLocale(locale: Locale): TextStyle =
+    when (locale.language.lowercase()) {
+        "zh" -> when (locale.country.lowercase()) {
+            "mo", "hk", "tw", "sg" -> TextStyle.SHORT // Traditional
```
Comment (marianeum) [inline backpack-common/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt:159]: can you also add a comment why we're doing this here?

Comment (bvitaliyg) [reply backpack-common/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt:159]: In addition to what we done in Kotlin Docs?

Comment (marianeum) [reply backpack-common/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt:159]: yeah - kotlin docs is for users modifying the format externally, but here for whoever reads the code and wonders why this is there

Comment (bvitaliyg) [reply backpack-common/src/main/java/net/skyscanner/backpack/calendar2/CalendarParams.kt]: Done

**Review by marianeum — APPROVED** [review]

**Pull request merged**